server/src/main/java/org/elasticsearch/search/rank/context/RankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -30,18 +32,37 @@ public abstract class RankFeaturePhaseRankCoordinatorContext {
     protected final int from;
     protected final int rankWindowSize;
     protected final boolean failuresAllowed;
+    protected final CustomRankInput customRankInput;
```

Comment:
I don't think this is the right place. This logic should live entirely on the text similarity rank builder.